### DISCUSSION TITLE
Using Azure communication services to replace smtp

### DISCUSSION THREAD

**Discussion by ghost** [OrchardCMS/OrchardCore#14531] [Q&A] [answered]
The current (default) email module uses smtp which has a different set of parameters than what's needed by azure "endpoint and token".   I'm currently making an altered version of the default email module to take the parameters.   My question is,  Is there a straight forward way to use my new module as the default for email notifications globally?

https://learn.microsoft.com/en-us/azure/communication-services/concepts/email/email-overview

**Comment by hishamco** [comment, marked as answer]
I already introduced different mail providers as modules in [OCC](https://github.com/OrchardCoreContrib/OrchardCoreContrib.Modules) simplify you can create a module and register your own `SmptService`

I might add `Azure Communication Services Email` if you don't wanna add it in OCC :) or it could be added here if all agree

Side note: we should consider the extensibility of the default abstractions APIs. That's why I prefer to push it to OCC first

**Reply by ghost** [reply]
Cool.  This is my first time seeing those modules, and it helps a lot.  The communicator resource also offers sms, so it would be useful for 2fa.  Still learning my way  around.

**Reply by hishamco** [reply]
Ya, I saw the SMS which I planned to integrate with if I have a time

So, let me know if you have interest to work on this alongside with other OCC email modules

**Comment by Piedone** [comment]
With https://github.com/OrchardCMS/OrchardCore/pull/15254 this is now part of OC and will be released in v1.9.
